src/kem/kem.h

```diff
@@ -70,6 +70,12 @@ extern "C" {
 #define OQS_KEM_alg_kyber_768 "Kyber768"
 /** Algorithm identifier for Kyber1024 KEM. */
 #define OQS_KEM_alg_kyber_1024 "Kyber1024"
+/** Algorithm identifier for ML-KEM-512-ipd KEM. */
+#define OQS_KEM_alg_ml_kem_512 "ML-KEM-512-ipd"
```

Comment:
Is it necessary to put the algorithm version into its name? We never did this until now. Or am I misunderstanding the rationale for the suffix "-ipd" as denoting the FIPS (draft level)? Why not leave the suffix away and update the implementation as and when the draft matures/changes (if necessary)? [NIST FIPS 203](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.203.ipd.pdf) also just uses the name(s) "ML-KEM-512" etc.

Reply:
We're not certain that there won't be algorithmic changes between the current FIPS draft and the final. While we have in the past just absorbed incompatible algorithm changes within the same function name, the idea here is to use distinct names -- "Kyber" for the current NIST Round 3 version, "IPD" for the current FIPS draft, and eventually just "ML-KEM" for the final FIPS draft.  If it turns out that IPD=ML-KEM, then we can easily solve that with some macros, but if that doesn't come to pass, we're better prepared for it.

Reply:
> If it turns out that IPD=ML-KEM, then we can easily solve that with some macros, 

True from the perspective of `liboqs` -- not so from the perspective of users of the library (and they are in significant majority): All code, documentation, write-ups and configs using "ML_KEM...-ipd" written between now and final standardization will have to be adapted as soon as the "-ipd" moniker gets dropped.

> but if that doesn't come to pass, we're better prepared for it.

In which way? If that happens, we have to change the `liboqs` code anyway. The work effort for all downstreams would be significantly eased if we use the final name already now, though: They (incl. "us" as in oqsprovider, etc.) at most need recompilation but no re-writing.

I always thought `liboqs` aims to aid in the adoption of PQC and not force all users to follow the nitty-gritty details of the standardization process. Wouldn't it be nice if people could rely on `liboqs` (as any changes during further standardization steps are kept under the hood)? I personally consider(ed) that one big advantage of using this library: All tests, use cases, documentation, papers could be retained throughout the NIST competition (for those algorithms that didn't change names or got dropped). Why drop this advantage now that final names are agreed-upon?

Reply:
> > If it turns out that IPD=ML-KEM, then we can easily solve that with some macros,
> 
> True from the perspective of `liboqs` -- not so from the perspective of users of the library (and they are in significant majority): All code, documentation, write-ups and configs using "ML_KEM...-ipd" written between now and final standardization will have to be adapted as soon as the "-ipd" moniker gets dropped.

The -ipd moniker doesn't need to be dropped, it can be kept as an alias.